File: Basic/Find all factorial numbers less than or equal to N/find-all-factorial-numbers-less-than-or-equal-to-n.py
#User function Template for python3

class Solution:
    def factorialNumbers(self, N):
        ans = []
    	def fact(n, fac, i):
    	    i+=1
    	    if fac>n:
    	        return
    	    ans.append(fac)
    	    fact(n, fac*i, i)
    	fact(N,1,1)
    	return ans


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__': 
    t = int(input())
    for _ in range(t):
        N = int(input())
        ob = Solution()
        ans = ob.factorialNumbers(N)
        for i in ans:
            print(i, end=" ")
        print()
        
# } Driver Code Ends

File: Difficulty: Easy/Find all factorial numbers less than or equal to n/find-all-factorial-numbers-less-than-or-equal-to-n.java
//{ Driver Code Starts
// Initial Template for Java
import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while (t-- > 0) {
            long N = sc.nextLong();

            Solution ob = new Solution();
            ArrayList<Long> ans = ob.factorialNumbers(N);
            for (long num : ans) {
                System.out.print(num + " ");
            }
            System.out.println();
        }
    }
}

// } Driver Code Ends


// User function Template for Java
class Solution {
    int cnt = 1;
    long fac = 1;
    ArrayList<Long> list = new ArrayList<>();
    ArrayList<Long> factorialNumbers(long n) {
        fac*=cnt;
        if(fac>n) {
            return list;
        }
        
        list.add(fac);
        cnt++;
        
        factorialNumbers(n);
        return list;
    }
}